static/scripts/assignment_table.js

```diff
@@ -0,0 +1,481 @@
+                const commentsPlugin = table.getPlugin('comments');
+
+                for (let i = 0; i < slicedData.length; i += 2) {
+                    const preferenceRow = slicedData[i];
+                    const assignmentRow = slicedData[i + 1];
+
+                    const userData = {
+                        researcher_id: preferenceRow.researchers.id,
+                        load_q1: preferenceRow.loadQ1,
+                        load_q2: preferenceRow.loadQ2,
+                        comment: preferenceRow.comment
+                    };
+
+                    const courseData = {};
+                    courses.forEach(course => {
+                        const col = table.propToCol(course.id);
+                        const comment = commentsPlugin.getCommentAtCell(i + lenFixedRowsText + 1, col);
+                        if (assignmentRow[course.id] !== '') {
+                            courseData[course.id] = {
+                                position: assignmentRow[course.id],
+                                comment: comment
+                            };
+                        }
+                    });
+
+                    const comments = {
+                        preference: preferenceRow.comment,
+                        assignment: assignmentRow.comment,
+                    };
+
+                    savedData.push({userData, courseData, comments});
+                }
+
+                const tableData = {
+                    data: savedData,
+                    isDraft: isDraft
+                };
+
+                try {
+                    const response = await fetch('/assignment/publish_assignments', {
+                        method: 'POST',
+                        headers: {
+                            'Content-Type': 'application/json',
+                        },
+                        body: JSON.stringify(tableData),
+                    });
+
+                    if (response.ok) {
+                        let msg = isDraft ? 'Draft saved' : 'Assignments published';
+                        updateToastContent(msg);
+                        toastNotification.show();
+                    } else {
+                        updateToastContent('Failed to publish assignments' + response.statusText);
+                        toastNotification.show();
+                    }
+                } catch (error) {
+                    updateToastContent('Error: ' + error);
+                    toastNotification.show();
+                }
+            }
+
+            $('#button-create-assignments').click(async function () {
+                await saveAssignment(true);
+            });
+
+            $('#button-clear-assignments').click(function () {
+                const newUsersRow = userRowsData(true);
+                data = fixedRows.concat(newUsersRow);
+                table.loadData(data);
+                updateToastContent('Data cleared');
+                toastNotification.show();
+            });
```

Comment:
The "clear assignment" button does clear the selections but not the values of the total per course, as well as put the title at it's default color.  

![image](https://github.com/user-attachments/assets/8e5d360f-b893-430c-a86c-580d7630693e)

Reply:
It is not a critical problem but it is something that should not be in a finished product.